ci: Move lint to be run on linux.20_04.16x runners

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #77290

We noticed that lint was taking a long time to run on github default
runners so we should utilize our own infra to run these jobs

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

seemethere committed:
ci: Move lint to be run on linux.2xlarge runners

We noticed that lint was taking a long time to run on github default
runners so we should utilize our own infra to run these jobs

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Move lint to be run on linux.2xlarge runners"


We noticed that lint was taking a long time to run on github default
runners so we should utilize our own infra to run these jobs

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

janeyx99 commented on .github/workflows/lint.yml:
```diff
@@ -9,15 +9,20 @@ on:
-      - name: Setup Python
-        uses: actions/setup-python@v2
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v2
```
ooh this is new

janeyx99 replied:
oh it didn't work in CI :/

seemethere committed:
Update on "ci: Move lint to be run on linux.2xlarge runners"


We noticed that lint was taking a long time to run on github default
runners so we should utilize our own infra to run these jobs

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Move lint to be run on linux.2xlarge runners"


We noticed that lint was taking a long time to run on github default
runners so we should utilize our own infra to run these jobs

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Move lint to be run on linux.2xlarge runners"


We noticed that lint was taking a long time to run on github default
runners so we should utilize our own infra to run these jobs

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Move lint to be run on linux.2xlarge runners"


We noticed that lint was taking a long time to run on github default
runners so we should utilize our own infra to run these jobs

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere commented:
This decrease lintrunner execution by ~1/2

* default (~13m30s): https://github.com/pytorch/pytorch/runs/6396090181?check_suite_focus=true
* changed (~7m): https://github.com/pytorch/pytorch/runs/6396368174?check_suite_focus=true

seemethere commented:
@pytorchbot merge this
